clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/view/AmazonClusterProvider.groovy

```diff
@@ -564,17 +565,17 @@ class AmazonClusterProvider implements ClusterProvider<AmazonCluster>, ServerGro
    */
   private static void updateServerGroupLaunchSettings(Map<String, AmazonServerGroup> serverGroups, Collection<CacheData> launchData) {
     for (ld in launchData) {
-      if (ld.relationships.containsKey(SERVER_GROUPS.ns)) {
-        def serverGroup = serverGroups[ld.relationships.serverGroups.first()]
-        if (serverGroup == null) {
-          continue
-        }
-
-        if (serverGroup.asg?.launchTemplate) {
-          def launchTemplateSpec = serverGroup.asg.launchTemplate as Map
-          serverGroup.launchTemplate = getLaunchTemplateVersion(ld, launchTemplateSpec.version as String)
-        } else {
-          serverGroup.launchConfig = ld.attributes
+      if (ld.relationships != null && ld.relationships.containsKey(SERVER_GROUPS.ns)) {
+        ld.relationships.serverGroups.each {
```

Comment:
I'd suggest `ld.relationships[SERVER_GROUPS.ns]` here since that is what the containsKey is checking on

Reply:
Good catch! thanks